NMS-12426: Add docs for BMP support (#2961)
OpenNMS/opennms: opennms-doc/guide-admin/src/asciidoc/text/telemetryd/protocols/bmp.adoc

==== BGP Monitoring Protocol

_BGP Monitoring Protocol (BMP)_ provides a convenient interface for obtaining route views.

> The integration in _{opennms-product-name}_ allows to utilize these route views, status updates and statistics for monitoring and management purposes.

Reviewer: ```suggestion
The integration in _{opennms-product-name}_ allows you to use these route views, status updates and statistics for monitoring and management.
```

[[telemetryd-bmp-parser]]
===== BMP TCP Parser

The BMP Parser accepts BMP connections from routers packets using a <<telemetryd-listener-tcp, TCP Listener>>.

====== Facts

[options="autowidth"]
|===
| Class Name          | `org.opennms.netmgt.telemetry.protocols.bmp.parser.BmpParser`
|===

====== Parameters

.Parameters for the BMP Parser
[options="header, autowidth"]
|===
| Parameter             | Description                                                                    | Required | Default value
| `dnsLookupsEnabled`   | Used to enable or disable DNS resolution of router and peer hostnames.         | no       | true
|===

[[telemetryd-bmp-statistics-adapter]]
===== BMP Telemetry Adapter

> The BMP Telemetry Adapter is used to handle _BMP_ statistic data received and parsed by the <<telemetryd-bmp-parser, BMP Parser>>.

Reviewer: ```suggestion
The BMP Telemetry Adapter handles _BMP_ statistic data received and parsed by the <<telemetryd-bmp-parser, BMP Parser>>.
```
What is "statistic" data? Can we just say "data"?

> Received statistics are associated as performance data to the router sending the statistics (which must exists as regular Node in _{opennms-product-name}_).

Reviewer: ```suggestion
Statistics received from the router are associated as performance data with that router. 
The router must exist as a regular node in _{opennms-product-name}_.
```

> To associate a router with the according _{opennms-product-name}_ Node, the IP address exporting BMP messages is used.

Reviewer: ```suggestion
{opennms-product-name} uses the IP address exporting BMP messages to associate a router with the particular _{opennms-product-name}_ node.
```
Not clear if this sentence is telling users to do this, or it's a statement of how OpenNMS does this by default.

> In addition, the Nodes Meta Data can be used to specify a BGP ID which will then be used to associate routers.

Reviewer: ```suggestion
In addition, the node's metadata can specify a BGP ID, which will then be used to associate routers.
```
Not clear if this should be a possessive: "the node's metadata". Please verify.

> If the parameter `metaDataNodeLookup` is not empty, it will be interpreted as a `context:key` meta data name which will be used to lookup a Node which has stored the queried BGP ID as value in exactly this key.

Reviewer: ```suggestion
If the parameter `metaDataNodeLookup` is not empty, it will be interpreted as a `context:key` metadata name, which will be used to lookup a node which has stored the queried BGP ID as a value in exactly this key.
```

====== Facts

[options="autowidth"]
|===
| Class Name          | `org.opennms.netmgt.telemetry.protocols.bmp.adapter.BmpTelemetryAdapter`
|===

====== Parameters

.Parameters for the BMP Telemetry Adapter
[options="header, autowidth"]
|===
| Parameter            | Description                                                                    | Required | Default value
| `metaDataNodeLookup` | Enables lookup using BGP IDs.                                                  | no       |
|===

[[telemetryd-bmp-peer-status-adapter]]
===== BMP Peer Status Adapter

> The BMP Peer Status Adapter is used to create events for peer status changes.

Reviewer: ```suggestion
The BMP Peer Status Adapter creates events for peer status changes.
```

> It handles _BMP_ Peer Up and Down messages received and parsed by the <<telemetryd-bmp-parser, BMP Parser>> and converts them to _{opennms-product-name}_ Events.

Reviewer: ```suggestion
It handles _BMP_ Peer Up and Down messages that the <<telemetryd-bmp-parser, BMP Parser>> receives and parses, and converts to _{opennms-product-name}_ events.
```

> The created events are associated to the router sending the messages (which must exists as regular Node in _{opennms-product-name}_).

Reviewer: ```suggestion
_{opennms-product-name}_ associates the created events with the router sending the messages.
This router must exist as regular node in _{opennms-product-name}_.
``

The events are called `uei.opennms.org/bmp/peerUp` and `uei.opennms.org/bmp/peerDown`.

> To associate a router with the according _{opennms-product-name}_ Node, the IP address exporting BMP messages is used.

Reviewer: ```suggestion
{opennms-product-name} uses the IP address exporting BMP messages to associate a router with the particular _{opennms-product-name}_ node.
```

> In addition, the Nodes Meta Data can be used to specify a BGP ID which will then be used to associate routers.

Reviewer: ```suggestion
In addition, the node's metadata can specify a BGP ID, which will then be used to associate routers.
```

> If the parameter `metaDataNodeLookup` is not empty, it will be interpreted as a `context:key` meta data name which will be used to lookup a Node which has stored the queried BGP ID as value in exactly this key.

Reviewer: ```suggestion
If the parameter `metaDataNodeLookup` is not empty, it will be interpreted as a `context:key` metadata name, which will be used to lookup a node which has stored the queried BGP ID as a value in exactly this key.
```

.Parameters for the BMP Parser
[options="header, autowidth"]
|===
| Parameter             | Description                                                                           | Required | Default value

> | `dnsLookupsEnabled`   | Used to enable or disable DNS resolution of router and peer hostnames.                | no       | true

Reviewer: ```suggestion
| `dnsLookupsEnabled`   | Enables or disables DNS resolution of router and peer hostnames.                | no       | true
```

| Parameter            | Description                                                                    | Required | Default value
| `metaDataNodeLookup` | Enables lookup using BGP IDs.                                                  | no       |
|===

[[telemetryd-openbmp-integration-adapter]]
===== OpenBMP Integration Adapter

> The OpenBMP Integration Adapter is used to integrate with an existing http://www.openbmp.org[OpenBMP] installation.

Reviewer: ```suggestion
The OpenBMP Integration Adapter integrates with an existing http://www.openbmp.org[OpenBMP] installation.
```

> It handles _BMP_ messages received and parsed by the <<telemetryd-bmp-parser, BMP Parser>> and creates _OpenBMP_ compatible messages which are then passed to the _OpenBMP_ kafka cluster.

Reviewer: ```suggestion
It handles _BMP_ messages the <<telemetryd-bmp-parser, BMP Parser>> receives and parses and creates _OpenBMP_-compatible messages, which are then passed to the _OpenBMP_ Kafka cluster.
```

This setup replaces the _Collector_ component of _OpenBMP_.

====== Facts

[options="autowidth"]
|===
| Class Name          | `org.opennms.netmgt.telemetry.protocols.bmp.adapter.openbmp.BmpIntegrationAdapter`
|===

====== Parameters

.Parameters for the OpenBMP Integration Adapter
[options="header, autowidth"]
|===
| Parameter            | Description                                                                    | Required | Default value

> | `kafka.*`            | Options used to create OpenBMP kafka producer. All known kafka settings are allowed but prefixed with `kafka.`. | no       |

Reviewer: ```suggestion
| `kafka.*`            | Uses options to create OpenBMP Kafka producer. Allows all known Kafka settings, but prefixed with `kafka.`. | no       |
```

> | `topicPrefix`        | Prefix used before each kafka topic                                            | no       |

Reviewer: ```suggestion
| `topicPrefix`        | Prefix used before each Kafka topic.                                            | no       |
```
